Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -96,45 +97,63 @@ protected override void ProcessRecord()
-                    lRequest.All = All;
-                    lRequest.Filter = Filter;
-                    do
-                    {
-                        DatasetList response = lRequest.Execute();
-                        if (response == null)
-                        {
-                            WriteError(new ErrorRecord(
-                                new Exception("List request to server responded with null."),
-                                "List request returned null", ErrorCategory.InvalidArgument, Project));
-                        }
-                        if (response.Datasets != null)
-                        {
-                            WriteObject(response.Datasets, true);
-                        }
-                        lRequest.PageToken = response.NextPageToken;
-                    }
-                    while (!Stopping && lRequest.PageToken != null);
+                    var datasets = DoListRequest(Project);
+                    WriteObject(datasets, true);
                     break;
                 case ParameterSetNames.Get:
-                    DatasetsResource.GetRequest gRequest = Service.Datasets.Get(Project, Dataset);
-                    try
-                    {
-                        var gResponse = gRequest.Execute();
-                        WriteObject(gResponse);
-                    }
-                    catch (GoogleApiException ex) when (ex.HttpStatusCode == HttpStatusCode.NotFound)
-                    {
-                        WriteError(new ErrorRecord(ex, 
-                            $"Error 404: Dataset {Dataset} not found in {Project}.",
-                            ErrorCategory.ObjectNotFound, 
-                            Dataset));
-                    }
+                    var dataset = DoGetRequest(Project, Dataset);
+                    WriteObject(dataset);
                     break;
                 default:
                     throw UnknownParameterSetException;
             }
         }
+
+        private IEnumerable<DatasetList.DatasetsData> DoListRequest(string project)
+        {
+            DatasetsResource.ListRequest lRequest = Service.Datasets.List(project);
+            lRequest.All = All;
+            lRequest.Filter = Filter;
+            do
+            {
+                DatasetList response = lRequest.Execute();
+                if (response == null)
+                {
+                    WriteError(new ErrorRecord(
+                        new Exception("List request to server responded with null."),
+                        "List request returned null", ErrorCategory.InvalidArgument, project));
+                }
+                if (response.Datasets != null)
+                {
+                    foreach(DatasetList.DatasetsData d in response.Datasets)
+                    {
+                        yield return d;
+                    }
+                    WriteObject(response.Datasets, true);
```

Comment:
Missed that.  Meant to cut it after I put in the for-each and then ran into the type issues above =/